core-feature-pack/src/main/resources/content/bin/common.ps1

```diff
@@ -0,0 +1,256 @@
+  $PROG_ARGS += "-Djboss.home.dir=$JBOSS_HOME"
+  $PROG_ARGS += "-jar"
+  $PROG_ARGS += "$JBOSS_HOME\jboss-modules.jar"
+  if ($MODULE_OPTS -ne $null){
+  	$PROG_ARGS += $MODULE_OPTS
+  }
+  $PROG_ARGS += "-mp"
+  $PROG_ARGS += "$JBOSS_MODULEPATH"
+  $PROG_ARGS += $entryModule
+  if ($serverOpts -ne $null){
+  	$PROG_ARGS += $serverOpts
+  }
+  return $PROG_ARGS
+}
+
+Function Process-Script-Parameters {
+Param(
+   [Parameter(Mandatory=$false)]
+   [string[]]$Params   
+
+) #end param
+    $res = @()
+	for($i=0; $i -lt $Params.Count; $i++){
+		$arg = $Params[$i]
+		if ($arg -eq '--debug'){
+			$global:DEBUG_MODE=$true
+			if ($args[$i+1] -match '\d+'){ #port number can only follow --debug
+				$global:DEBUG_PORT = $Params[$i+1]
+				$i++
+				continue
+			}
+		}elseif ($arg -contains '-Djava.security.manager'){
+			echo "ERROR: The use of -Djava.security.manager has been removed. Please use the -secmgr command line argument or SECMGR=true environment variable."
+			exit
+		}elseif ($arg -eq '-secmgr'){
+			$global:SECMGR = $true
+		}elseif ($arg -eq '--background'){
+			$global:RUN_IN_BACKGROUND = $true
+		}else{
+			$res+=$arg
+		}
+	}
+	return $res
+}
+
+Function Start-WildFly-Process {
+ Param(
+   [Parameter(Mandatory=$true)]
+   [string[]] $programArguments,
+   [boolean] $runInBackground = $false   
+
+) #end param
+
+	if(($JBOSS_PIDFILE -ne '') -and (Test-Path $JBOSS_PIDFILE)) {
+		$processId = gc $JBOSS_PIDFILE
+		if ($processId -ne $null){
+			$proc = Get-Process -Id $processId -ErrorAction SilentlyContinue
+		}
+		if ($proc -ne $null){
+			Write-Warning "Looks like a server process is already running. If it isn't then, remove the $JBOSS_PIDFILE and try again"
+			return
+		}else{
+			Remove-Item $JBOSS_PIDFILE
+		}
+	}
+
+	if($runInBackground) {	
+		$process = Start-Process -FilePath $JAVA -ArgumentList $programArguments -NoNewWindow -RedirectStandardOutput $global:CONSOLE_LOG -WorkingDirectory $JBOSS_HOME -PassThru
+		$processId = $process.Id;
+		echo "Started process in background, process id: $processId" 
+		if ($JBOSS_PIDFILE -ne $null){
+			$processId >> $JBOSS_PIDFILE  	
+		}
+	} else {
+		try{
+			pushd $JBOSS_HOME
+			& $JAVA $programArguments
+		}finally{
+			pushd $SCRIPTS_HOME
+		}
+	}
+
+}
+
+Function Set-Global-Variables {
+PARAM(
+[Parameter(Mandatory=$true)]
+   [string]$baseDir
+)
+
+	# determine the default base dir, if not set
+	$global:JBOSS_BASE_DIR = $baseDir;
+	
+	# determine the default log dir, if not set
+	$global:JBOSS_LOG_DIR = Get-Env JBOSS_LOG_DIR $JBOSS_BASE_DIR\log
+	
+	# determine the default configuration dir, if not set
+	$global:JBOSS_CONFIG_DIR = Get-Env JBOSS_CONFIG_DIR $JBOSS_BASE_DIR\configuration
+	
+	$global:CONSOLE_LOG = $JBOSS_LOG_DIR + '\console.log'
+}
+
+Function Set-Global-Variables-Standalone {
+	$dir = Get-Env JBOSS_BASE_DIR $JBOSS_HOME\standalone
+	Set-Global-Variables -baseDir $dir
+}
+
+Function Set-Global-Variables-Domain {
```

Comment:
it is a function, that is only called when needed, basically just domain.ps1 calls it.